src/all/deviantart/src/eu/kanade/tachiyomi/extension/all/deviantart/DeviantArt.kt

```diff
@@ -0,0 +1,186 @@
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.parser.Parser
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class DeviantArt : HttpSource() {
+    override val name = "DeviantArt"
+    override val baseUrl = "https://deviantart.com"
+    override val lang = "all"
+    override val supportsLatest = false
+
+    private val backendBaseUrl = "https://backend.deviantart.com"
+    private fun backendBuilder() = backendBaseUrl.toHttpUrl().newBuilder()
+
+    private val dateFormat by lazy {
+        SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss zzz", Locale.ENGLISH)
+    }
+
+    private fun parseDate(dateStr: String): Long {
+        return try {
+            dateFormat.parse(dateStr)!!.time
+        } catch (_: ParseException) {
+            0L
+        }
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        throw UnsupportedOperationException(SEARCH_ERROR_MSG)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        throw UnsupportedOperationException(SEARCH_ERROR_MSG)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = backendBuilder()
+            .addPathSegment("rss.xml")
+            .addQueryParameter("q", query)
+            .build()
+
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoupXml()
+        requireNotNull(document.selectFirst("item")) { SEARCH_ERROR_MSG }
+
+        val (username, folderId) = response.request.url.queryParameter("q")!!.split(" ")
+            .find { it.startsWith("gallery:") }
+            ?.substringAfter(":")
+            ?.split("/")
+            ?: throw IllegalArgumentException(SEARCH_ERROR_MSG)
+
+        val newRequest = GET("$baseUrl/$username/gallery/$folderId", headers)
+        val newResponse = client.newCall(newRequest).execute()
+        val manga = mangaDetailsParse(newResponse)
+
+        return MangasPage(listOf(manga), false)
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        throw UnsupportedOperationException()
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        throw UnsupportedOperationException()
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val document = response.asJsoup()
+        val manga = SManga.create().apply {
+            // If manga is sub-gallery then use sub-gallery name, else use gallery name
+            title = document.selectFirst("._2vMZg + ._2vMZg")?.text()?.substringBeforeLast(" ")
+                ?: document.selectFirst(".kyUNb")!!.ownText()
+            author = document.title().substringBefore(" ")
+            description = document.selectFirst(".py0Gw._3urCH")?.wholeText()
+            thumbnail_url = document.selectFirst("._1xcj5._1QdgI img")!!.attr("src")
```

Comment:
Use `absUrl`, also, `thumbnail_url` is not critical so you should remove the non-null assert.